fsm/fsm.go

```diff
@@ -87,7 +87,8 @@ func (d fsmHandler) Plan(events []statemachine.Event, user interface{}) (interfa
 	if _, ok := eventName.(nilEvent); ok {
 		return nil, 1, nil
 	}
-	if err != nil {
+	_, justRecord := err.(justRecordError)
```

Comment:
I find using errors as sentinel values a little confusing.

Reply:
noted -- I'm leaving as is but maybe you are right and may revisit.